### PR TITLE
Sidebar state

### PR DESCRIPTION
**Pull request by UladBohdan**
`GET /api/storage/sidebar` to get sidebar state, `POST /api/storage/sidebar` to update it

`initAll.js` script is improved to return rich main page state

Updated `main-page` sample: https://github.com/babajka/gists/blob/master/main-page-state.js
Created `sidebar` sample: https://github.com/babajka/gists/blob/master/sidebar-state.js

### CODE REVIEW

**Comment by UladBohdan** [comment]
[to store sidebar state](https://trello.com/c/wnqWLGlt/350-to-store-sidebar-state)

File: src/api/storage/controller.js [hunk 1]
```diff
@@ -15,8 +17,10 @@ const MAIN_PAGE_ENTITIES_QUERIES = {
   topics: () => Topic.getAll(),
 };
 
-export const getMainPage = ({ user }, res, next) =>
-  StorageEntity.getValue(MAIN_PAGE_KEY)
+const SIDEBAR_ENTITIES_QUERIES = pick(MAIN_PAGE_ENTITIES_QUERIES, ['tags']);
```
Comment (Drapegnik) [inline src/api/storage/controller.js]: why not just `MAIN_PAGE_ENTITIES_QUERIES.tags`?

File: src/api/storage/controller.js [hunk 2]
```diff
@@ -15,8 +17,10 @@ const MAIN_PAGE_ENTITIES_QUERIES = {
...
+
+const getState = (user, storageKey, entitiesQueries, includeLatestArticles) =>
```
Comment (Drapegnik) [inline src/api/storage/controller.js]: lets use params destructuring:
```js
getState = ({ user, storageKey, entitiesQueries, includeLatestArticles = false })
```

```js
getState({ storageKey: 2, includeLatestArticles: true })
```

much better than:

```js
getState(2, param, foo, false)
```

File: src/db/scripts/initAll.js [hunk 3]
```diff
@@ -199,20 +203,103 @@ const initDiaries = () =>
...
+      const tagsByTopic = {};
+      TOPIC_SLUGS.forEach(topicSlug => {
+        tagsByTopic[topicSlug] = filter(tags, { topic: topicsBySlug[topicSlug]._id }).map(
+          ({ _id }) => _id
+        );
+      });
+
+      const articlesByTag = {};
+      tags.forEach(tag => {
+        articlesByTag[tag.slug] = filter(articles, art =>
```
Comment (Drapegnik) [inline src/db/scripts/initAll.js]:
```suggestion
        articlesByTag[tag.slug] = articles.filter(art =>
```

File: src/db/scripts/initAll.js [hunk 4]
```diff
@@ -199,20 +203,103 @@ const initDiaries = () =>
...
+const initSidebarState = metadataTestingUser =>
+  Tag.find()
+    .then(async tags => {
+      const topics = await Topic.find().exec();
+      const topicsBySlug = keyBy(topics, 'slug');
+
+      const blocks = ['themes', 'personalities', 'times', 'locations', 'brands', 'authors'].map(
+        topicSlug => ({
+          topic: topicSlug,
+          tags: filter(tags, { topic: topicsBySlug[topicSlug]._id }).map(tag => tag._id),
```
Comment (Drapegnik) [inline src/db/scripts/initAll.js]: `.filter`

File: src/api/storage/controller.js (hunk 1 above)
Comment (UladBohdan) [reply src/api/storage/controller.js]: `MAIN_PAGE_ENTITIES_QUERIES.tags` returns the only function associated with `tags`.
`pick()` preserves the structure: we get `{ tags: func() }`

**Pull request merged**